spr/src/commit_message.rs

```diff
@@ -0,0 +1,426 @@
+// Write is needed by Command::Stdin.write_all()
+use std::io::Write;
+use std::process::{Command, Stdio};
+
+use crate::{
+    error::{Error, Result},
+};
+
+// Notice: use BTreeMap to make it easier to iterate trailer keys in order.
+pub type TrailerMap = std::collections::BTreeMap<String, Vec<String>>;
+
+#[derive(Debug, PartialEq)]
+pub struct CommitMessage {
+
+    /// Subject of the message (i.e. very first line)
+    pub subject:  String,
+
+    /// Body of the message, *EXCLUDING* the contents of the trailers
+    /// section. Empty string if no body.
+    pub body:     String,
+
+    /// Map of trailer keys to trailer values (e.g, "key: value...").
+    pub trailers: TrailerMap,
+}
+
+impl CommitMessage {
+
+    pub fn render(&self) -> String {
+        let mut ret: String = "".to_string();
+
+        ret.push_str(&format!("{}\n", &self.subject));
```

Comment:
Do we want to add a default subject if it's empty?

trying with git:
- if the entire commit is empty, it'll fail the commit
- if the commit msg isn't empty but starts with a blank line for the subject, then the _entire_ body becomes the subject

Reply:
added "MISSING COMMIT MESSAGE SUBJECT"